### PR TITLE
TRAIN-293: creating the login auth

### PR DESCRIPTION
**Pull request by Stanley-Maraba**

### CODE REVIEW

File: src/main/java/com/budgetapp/controllers/UserController.java [hunk 1]
```diff
@@ -48,4 +49,14 @@ public ResponseEntity<User> findUserById(@PathVariable final Long id) {
         final User user = userService.findUserById(id);
         return ResponseEntity.ok(user);
     }
+
+    @PostMapping(path = "/login", produces = MediaType.TEXT_PLAIN_VALUE)
+    public ResponseEntity<String> login(@RequestParam final String username, @RequestParam final String password) {
+            final User Authuser = userService.authenticatingUser(username, password);
```
Comment (JOlsen-Neslo) [inline src/main/java/com/budgetapp/controllers/UserController.java]: your variable name does not follow Java standards

File: src/main/java/com/budgetapp/controllers/UserController.java [hunk 2]
```diff
@@ -48,4 +49,14 @@ public ResponseEntity<User> findUserById(@PathVariable final Long id) {
         final User user = userService.findUserById(id);
         return ResponseEntity.ok(user);
     }
+
+    @PostMapping(path = "/login", produces = MediaType.TEXT_PLAIN_VALUE)
```
Comment (JOlsen-Neslo) [inline src/main/java/com/budgetapp/controllers/UserController.java]: we shouldn't be producing plain text

File: src/main/java/com/budgetapp/controllers/UserController.java [hunk 3]
```diff
@@ -48,4 +49,14 @@ public ResponseEntity<User> findUserById(@PathVariable final Long id) {
...
+    public ResponseEntity<String> login(@RequestParam final String username, @RequestParam final String password) {
```
Comment (JOlsen-Neslo) [inline src/main/java/com/budgetapp/controllers/UserController.java]: If you mark this endpoint to consume JSON, you can create an object that should be used as the request body. Based on this you are expecting form data

File: src/main/java/com/budgetapp/controllers/UserController.java [hunk 4]
```diff
@@ -48,4 +49,14 @@ public ResponseEntity<User> findUserById(@PathVariable final Long id) {
...
+            final User Authuser = userService.authenticatingUser(username, password);
+        if (Authuser != null) {
+            return ResponseEntity.ok("Successful");
```
Comment (JOlsen-Neslo) [inline src/main/java/com/budgetapp/controllers/UserController.java]: A 200 response means it was successful, the message here is redundant

File: src/main/java/com/budgetapp/controllers/UserController.java [hunk 5]
```diff
@@ -48,4 +49,14 @@ public ResponseEntity<User> findUserById(@PathVariable final Long id) {
...
+        } else {
+            return ResponseEntity.notFound().build();
```
Comment (JOlsen-Neslo) [inline src/main/java/com/budgetapp/controllers/UserController.java]: General API standards for not being authenticated should return a 401 error not a 404

File: src/main/java/com/budgetapp/services/UserService.java [hunk 6]
```diff
@@ -32,4 +32,9 @@ public User findUserById(final Long id) {
     public void deleteUser(final Long id) {
         userRepository.deleteById(id);
     }
+
+    public User authenticatingUser(final String username, final String password) {
+        final User user = userRepository.findByUsernameAndPassword(username, password).orElseThrow(() -> new EntityNotFoundException(String.format("")));
```
Comment (JOlsen-Neslo) [inline src/main/java/com/budgetapp/services/UserService.java]: Your message is completely pointless?

```suggestion
        final User user = userRepository.findByUsernameAndPassword(username, password).orElseThrow(EntityNotFoundException::new);
```

**Review by JOlsen-Neslo — CHANGES_REQUESTED** [review]

File: src/main/java/com/budgetapp/services/UserService.java (hunk 6 above)
Comment (Stanley-Maraba) [reply src/main/java/com/budgetapp/services/UserService.java]: .orElse(null) used this instead as EntityNotFoundException is a custom exception that returns 404

**Review by JOlsen-Neslo — APPROVED** [review]

**Pull request merged**